### ISSUE TITLE
Mailchimp API support

### ISSUE THREAD

**Issue by geelen** [fab-spec/fab#16]
Thanks to @gragland for the use case. Divjoy generates a CRA project with a server with two dependencies: `aws-sdk` (which works fine) and `mailchimp-api-v3`, which doesn't.

Two dependencies are the problem:

https://www.npmjs.com/package/request crashes requiring `tls`
https://www.npmjs.com/package/tar depends on `fs`

Injecting `memfs` for `fs` makes tar compile, but `request` is a little more difficult.

Feels to me that there might be a library we could find/make that has the same API as `request` but backs onto `fetch` instead.

_Or_, Divjoy could call the Mailchimp API without using the `mailchimp-api-v3`, since it's only using it in the following lines:

```js
// Add your Mailchimp credentials here
const apiKey = "";
const listId = "";
const mailchimp = new Mailchimp(apiKey);

app.post("/api/newsletter/subscribe", function(req, res) {
  mailchimp
    .request({
      method: "POST",
      path: "/lists/" + listId + "/members",
      body: {
        email_address: req.body.email,
        // Set status to "subscribed" to disable double-opt-in
        status: "pending"
      }
    })
    .then(result => {
      res.send({ status: "success" });
    })
    .catch(err => {
      res.send({ status: "error" });
    });
});
```

@gragland, thoughts?

**Comment by geelen** [comment]
Oh, mailchimp's auth is really simple? From their library:

```js
this.__api_key = api_key;
this.__base_url = "https://"+ this.__api_key.split('-')[1] + ".api.mailchimp.com/3.0"

// ...
    request({
      method : method,
      url : mailchimp.__base_url + path,
      auth : {
        user : 'any',
        password : mailchimp.__api_key
      },
      json : body,
      qs : query,
      headers : headers,
    }, function (err, response) {

// ...
```

That looks reasonable to just merge into the above usage and use `fetch` directly and drop the dependency on `mailchimp-api-v3`

**Comment by gragland** [comment]
Normally swapping out Mailchimp like that would be a non-issue, but people are using Divjoy to export code that they intend to build on and they’d potentially be adding new dependencies that could also be problematic. I guess I’m not yet sure whether making deployment easier trumps that potential confusion. Maybe 🤔. Any sense of whether it doable to patch all the potential node APIs? I vaguely recall that the CodeSandbox team had to do this for their clientside webpack replacement.

**Comment by geelen** [comment]
Yeah certainly having a `request` module that "just worked" inside a FAB
would be better, it'd mean that `@fab/compile` could include that by
default for all new projects. Turning on `fs`: `memfs` alias seems like a
good plan too.

Know anyone at codesandbox we could ask for guidance here?

**Comment by gragland** [comment]
I think that work was done by Ives (https://twitter.com/compuives). Don't know him personally, but he's always been pretty responsive. Maybe he's at React Rally as well?

There's also https://github.com/codesandbox/codesandbox-client/tree/master/standalone-packages/codesandbox-browserfs although sounds like `fs` isn't the hard part.

I don't yet fully understand what FAB is doing under the hood, so maybe this is a dumb question, but what actual environment is it running in, if not Node? Is it possible to actually compile with Node or take advantage of a project that already does (Zeit's Builders system?). Again, I'm totally naive when it comes to all things bundling and compiling so this might not be useful :)
